forge/routes/api/deviceLive.js

```diff
@@ -38,6 +38,13 @@ module.exports = async function (app) {
             })
             return
         }
+        if (request.body.settingsHash && request.body.settingsHash !== (request.device.settingsHash)) {
+            reply.code(409).send({
+                error: 'incorrect-settings',
+                settingsHash: request.device.settingsHash
+            })
+            return
+        }
```

Comment:
```suggestion
        if (request.body.settings && request.body.settings !== (request.device.settingsHash)) {
            reply.code(409).send({
                error: 'incorrect-settings',
                settings: request.device.settingsHash,
                snapshot: request.device.targetSnapshot?.hashid || null
            })
            return
        }
```

On the external API, I think we should call this `settings` not `settingsHash` - we don't use `snapshotHash`.

I also think the response should include both the snapshot and settings values. My suggested change updates one case, but GitHub won't let me extend my suggested change into code you haven't modified - so the response in line 35-37 should be updated to include `settings`.

That will then allow the device to be more selective over whether it grabs both things each time, or just the one that has changed.